src/main/java/de/hysky/skyblocker/mixins/AbstractContainerMenuMixin.java

```diff
@@ -17,5 +18,6 @@ private void onSetStackInSlot(int slot, int revision, ItemStack stack, CallbackI
 			InventorySearch.refreshSlot(slot);
 		}
 		ItemPickupWidget.getInstance().onItemPickup(slot, stack);
+		((ItemStackUpdateDurability) (Object) stack).skyblocker$getAndCacheDurability();
```

Comment:
Don’t you need this to be called on component update as well? We already have a mixin for that somewhere as well.

Reply:
Are you referring to the code for `cancelComponentUpdateAnimation`? That occurs in renderer code: `ItemInHandRendererMixin`.
There is also `DataComponentHolderMixin`, but that method doesn't seem to have a relevant update function.
Also, I don't believe the server even has a way to partially update an inventory stack's components: the only relevant packet seems to be `ClientboundContainerSetSlotPacket`, which always goes through `setItem`.

Reply:
(I also did verify that the durability bar correctly updates while mining.)